internal/engine/eval_status.go

```diff
@@ -129,71 +129,22 @@ func (e *executor) createOrUpdateEvalStatus(
 		return nil
 	}
 
-	// Upsert evaluation
```

Comment:
These DB queries need to happen after the write to the evaluation history table since we are now tracking the rule-entity ID. They also now are run in the same transaction as the other queries.